cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -71,10 +71,10 @@ class BufferExpander
 
     BufferExpander(std::shared_ptr<ConcreteBuffer> const & buf, bool clone, ctor_passkey const &)
```

Comment:
You need to call the constructor of the parent class!

Reply:
fixed.

Reply:
I don't think it's possible, due to `when initialized here [-Werror=reorder]`

`m_begin` and `m_end` need to rely on `m_data`, however the `m_data`'s type are different in the derived class, which result that we cannot put `m_data` into the base class.

so the order need to be:
- derived class `m_data` initialized
- `m_start` and `m_end` initialized.

if we use parent constructor, then the compiler will complain that the base class member needs to be initialized first. so, unless we turn off the "reorder warning", we cannot use parent constructor.

Reply:
We must obey the initialization order.  It's not an option to initialize member data without following the order they are declared.

The type of derived class `m_data` probably does not matter.  What matters is their address.  Since `m_data` is after `m_begin` and `m_end`, the address of `m_data` cannot be known when `m_begin` and `m_end` appears in the initialization list.

Thus, `m_begin` and `m_end` need to be overridden after the derived classes are constructed.  We cannot initialize `m_begin` and `m_end` from the derived class initialization list through constructor delegate.

Reply:
sure, fixed.